### PR TITLE
User Tracking

### LINKED ISSUE #298: functional test for userprofile.usertracking()
The user tracking feature of IRCLogParser provides the following results:

1. user to channel graph.
1. user nick change graph.
1. keyword digest

We need to have a test to verify the proper functioning of this feature in IRCLogParser software. The functional tests must use the real Ubuntu logs of one month in duration and must use the real config module for testing.

The newly written test would go to `test/functional-test/userprofile.py` module.

### PR DESCRIPTION
**Pull request by Kausam**
## What? Why?
Fix #298  . (Add Reference to the issue if any)

Changes proposed in this pull request:
- add functional test for user tracking

## Checks
- [ ] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Make sure you have added required documentation
- [ ] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@kaushik-rohit
@prasadtalasila

### CODE REVIEW

**Comment by kaushik-rohit** [comment]
@prasadtalasila @Kausam The PR looks good to me. Only one thing, some of the tests are already there in test_functional.py and might need to be removed.

File: test/functional-test/data/output/2013/01/network/degree_node_number_csv/in_degree [hunk 1]
```diff
@@ -0,0 +1,263 @@
+(lp0
```
Comment (prasadtalasila) [inline test/functional-test/data/output/2013/01/network/degree_node_number_csv/in_degree]: do we need these data files? Can't we reuse the existing data files in tests/data?

File: test/functional-test/test_userprofile.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import networkx as nx
+import lib.util as util
+import lib.config as config
+import os
+import StringIO
+import sys
+from networkx.algorithms.components.connected import connected_components
+from numpy.testing import assert_array_equal
+
+class UserTrackingFunctionalTest(unittest.TestCase):
```
Comment (prasadtalasila) [inline test/functional-test/test_userprofile.py]: As mentioned in issue #298, we only need two functional tests here. One for user to channel tracking and another is to user nick change graph. The test functions of network module do not belong here.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@Kausam please see the requested changes. Also as mentioned by @kaushik-rohit , if there is a similar test in [test_functional.py](https://github.com/prasadtalasila/IRCLogParser/blob/master/test/functional-test/test_functional.py), then the code should be removed from [test_functional.py](https://github.com/prasadtalasila/IRCLogParser/blob/master/test/functional-test/test_functional.py)

**Comment by prasadtalasila** [comment]
@Kausam you are partially correct on having HITS test in user tracking. As per the paper, Fig 11 (page 19), TF-IDF vs HITS comparison is part of user profile.

File: test/functional-test/test_userprofile.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import lib.util as util
+import lib.config as config
+import os
+import StringIO
+import sys
+import filecmp
+from networkx.algorithms.components.connected import connected_components
+from numpy.testing import assert_array_equal
+
+class UserTrackingFunctionalTest(unittest.TestCase):
```
Comment (prasadtalasila) [inline test/functional-test/test_userprofile.py]: Can we change the class name to "UserProfileTest"

File: test/functional-test/test_userprofile.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        message_num_graph, top_hub, top_keyword_overlap, top_auth = network.identify_hubs_and_experts(self.log_data, nicks, nick_same_list)
+        
+        sys.stdout = sys.__stdout__
+        capturedOutput.close()
+
+        util.save_to_disk(top_hub, self.out_dir + "top_hub")
+        util.save_to_disk(top_keyword_overlap, self.out_dir + "top_keyword_overlap")
+        util.save_to_disk(top_auth, self.out_dir + "top_auth")
+        util.save_to_disk(message_num_graph, self.out_dir + "message_num_graph")
+
```
Comment (prasadtalasila) [inline test/functional-test/test_userprofile.py]: assertions section is missing in this test.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@Kausam please see the requested changes. Where possible, the existing test data must be reused.

**Review by prasadtalasila — APPROVED** [review]

**Pull request merged**